internal/plugin/plugin.go

```diff
@@ -11,19 +12,18 @@ import (
 	"strconv"
 	"strings"
 
+	"github.com/bufbuild/buf/private/bufpkg/bufplugin/bufpluginconfig"
 	"github.com/sethvargo/go-envconfig"
 	"golang.org/x/mod/semver"
 	"gopkg.in/yaml.v3"
 )
 
 // Plugin represents metadata (and filesystem path) information about a plugin.
 type Plugin struct {
-	Path        string       `yaml:"-"`
-	Relpath     string       `yaml:"-"`
-	Name        string       `yaml:"name"`
-	Version     string       `yaml:"plugin_version"`
-	Deps        []Dependency `yaml:"deps,omitempty"`
-	DefaultOpts []string     `yaml:"default_opts,omitempty"`
+	Path    string `yaml:"-"`
+	Relpath string `yaml:"-"`
+	// Parsed external yaml config
+	bufpluginconfig.ExternalConfig `yaml:"-"`
```

Comment:
Instead of duplicating reading from yaml I'm just embedding the struct we use in buf's CLI here (this is just for tests/builds).